modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -187,15 +206,62 @@ struct IEUnit {
     }
 
     // This method is [supposed to be] called at Island compilation stage
+    // TODO: Move to a new OpenVINO Core API!
     cv::gimpl::ie::IECompiled compile() const {
         auto this_plugin = IE::PluginDispatcher().getPluginByDevice(params.device_id);
+
+        // Load extensions (taken from DNN module)
+        if (params.device_id == "CPU" || params.device_id == "FPGA")
+        {
+            const std::string suffixes[] = { "_avx2", "_sse4", ""};
+            const bool haveFeature[] = {
+                cv::checkHardwareSupport(CPU_AVX2),
+                cv::checkHardwareSupport(CPU_SSE4_2),
+                true
+            };
+            std::vector<std::string> candidates;
+            for (auto &&it : ade::util::zip(ade::util::toRange(suffixes),
+                                            ade::util::toRange(haveFeature)))
+            {
+                std::string suffix;
+                bool available = false;
+                std::tie(suffix, available) = it;
+                if (!available) continue;
+#ifdef _WIN32
+                candidates.push_back("cpu_extension" + suffix + ".dll");
+#elif defined(__APPLE__)
+                candidates.push_back("libcpu_extension" + suffix + ".so");  // built as loadable module
+                candidates.push_back("libcpu_extension" + suffix + ".dylib");  // built as shared library
+#else
+                candidates.push_back("libcpu_extension" + suffix + ".so");
+#endif  // _WIN32
+            }
+            for (auto &&extlib : candidates)
+            {
+                try
+                {
+                    this_plugin.AddExtension(IE::make_so_pointer<IE::IExtension>(extlib));
+                    CV_LOG_INFO(NULL, "DNN-IE: Loaded extension plugin: " << extlib);
+                    break;
+                }
+                catch(...)
+                {
+                    CV_LOG_WARNING(NULL, "Failed to load IE extension " << extlib);
```

Comment:
Warning is too noise here (on each candidate).
Also `cpu_extension` loading will not be needed anymore in future OV releases.

Reply:
Ok will rewrite once we move to "Core API"